Directory.Packages.props

```diff
@@ -68,14 +68,14 @@
     <PackageVersion Include="Microsoft.AspNetCore.TestHost" Version="7.0.9" />
     <PackageVersion Include="Microsoft.AspNetCore.Mvc.Testing" Version="7.0.9" />
     <PackageVersion Include="Microsoft.CodeAnalysis.PublicApiAnalyzers" Version="[3.11.0-beta1.23402.2]" />
-    <PackageVersion Include="Microsoft.CodeCoverage" Version="[17.6.3]" />
+    <PackageVersion Include="Microsoft.CodeCoverage" Version="[17.7.2,18.0.0)" />
```

Comment:
What I noticed working on https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1347 is that `Microsoft.NET.Test.Sdk` now brings in `Microsoft.CodeCoverage`. What I think we should do is...

* Remove this line: https://github.com/open-telemetry/opentelemetry-dotnet/blob/fe78453c03feb8dbe506b2a0284312bdfa1367c5/build/Common.props#L32
  Kind of strange we were ever adding this to all projects in the first place 🤣 
* Remove the version specified in this props file (it shouldn't be needed).

I have those two changes on my contrib PR and it seems to be working fine.

Reply:
I have executed locally all step, looks good after changes: 2c20e9a110e70ce745aff6561f7e2fac56f6812e